plugins/parsers/prometheus/parser.go

```diff
@@ -28,12 +33,28 @@ func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	// Read raw data
 	buffer := bytes.NewBuffer(buf)
 	reader := bufio.NewReader(buffer)
+	mediatype, params, err := mime.ParseMediaType(p.Header.Get("Content-Type"))
```

Comment:
Could you please move this line to directly before the `if err...` (line 39) so we directly see that the error correspond to the media-type determination part!

Reply:
Done